packages/block-library/src/social-link/editor.scss

```diff
@@ -1,19 +1,33 @@
-// The editor uses the button component, the frontend uses a link.
-// Therefore we unstyle the button component to make it more like the frontend.
 .wp-block-social-links .wp-social-link {
 	line-height: 0;
+}
+
+// The editor uses the button element, the frontend uses a link.
+// Therefore we unstyle the button element to make it more like the frontend.
+.wp-block-social-link-anchor {
```

Comment:
Would we be able to lower the specificity here? If we ever want to support border or typography styling for this block, these styles could interfere.

Reply:
I was planning to look at dropping the specificity here over in https://github.com/WordPress/gutenberg/pull/61032 as the block style variations need it as well